Zend/tests/partial_function/reflection-resolver.phpt

```diff
@@ -0,0 +1,33 @@
+--TEST--
+Partial Reflection Resolver
+--FILE--
+<?php
+
+function volume(int $x, int $y, int $z): int {
+    return $x * $y * $z;
+}
+
+class Resolver {
+    private array $context = ['x' => 3];
+
+    public function resolve(callable $c) {
+        $f = new ReflectionFunction(Closure::fromCallable($c));
```

Comment:
Closure::fromCallable is not mandatory, right?

Reply:
It is in this case, because you need to fold arbitrary callables into a common format.  resolve() here doesn't work with just partials, although it always returns a partial.

Reply:
$c(?) would work too IIUC

Reply:
This is how I understood it as well, having a callable $c, the result of these expressions will be equivalent:
```php
Closure::fromCallable($c);
$c(?);
```
Will there be any performance difference between them?

Also, if $c is already a Closure, `fromCallable()` will just return it: https://3v4l.org/Debh5
@krakjoe, I was wondering if it would be possible to have the same optimization from the partial function syntax.

Reply:
that's already the case :)

![image](https://user-images.githubusercontent.com/29315886/118259642-4ed53380-b4a9-11eb-8fa1-990cce2bc2cd.png)

Reply:
> $c(?) would work too IIUC

yes

![image](https://user-images.githubusercontent.com/29315886/118260587-6bbe3680-b4aa-11eb-8c5b-f9e8e406b976.png)

Reply:
> that's already the case :)

Sorry I didn't express my question well enough.
I meant that if `$c` is a Closure, `$c(?)` to return the same object making `$c === $c(?)`

The context being that we can then replace all instances of `Closure::fromCallable($callable)` with `$callable(?)` without the performance impact of creating a new Closure.
Other than this, I'm also curios if there are other performance differences we could expect between the two ways of creating a Closure from a callable.

Reply:
Now that is a fascinating use case I had not though of.  Essentially `$callable(?)` becomes an alias for `Closure::fromCallable()`?  neat. :-)  We'd have to benchmark the difference later to see if it's even measurable.  As long as it's a least a wash, that's a nice bonus.

Reply:
@Crell Hmm, it means, more tests should be added, like this one (to be sure this won't break after later optimizations):

```php
class C {
    public function getPrivateCaller(): callable {
        return $this->privateMethod(?);
    }

    private function privateMethod(): void {
        echo "Called C#privateMethod()";
    }
}

(new C)->getPrivateCaller()();
```